### DISCUSSION TITLE
API is failing in server components but working in client components

### DISCUSSION THREAD

**Discussion by vinitgundeti** [vercel/next.js#69998] [Help]
### Link to the code that reproduces this issue

https://github.com/vinitgundeti/peps-api-fail

### To Reproduce

Create a server component.
Call api in server component

**Server Component**
```
export default async function Home() {
  const response = await apiCall('/get-home-content')
  const homeData = response?.data?.[0];
  return (
    homeData ? <h1>Welcome</h1> : <h1>Error</h1>
  );
}
```

**apiCall Function**

```
export const apiCall = async (
  slug,
  method = "GET",
  body = null,
  token,
  headers = {}
) => {
  try {
    const options = {
      cache: "no-store",
      method,
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
        Authorization: "Bearer " + token,
        ...headers,
      },
    };

    if (body) {
      options.body = JSON.stringify(body);
    }
    try {
      const response = await fetch(`${BASE_URL + slug}`, options);
      if (response.status !== 200) {
        const error = await response.json();
        console.log("error : ", error);
        return error;
      }

      const data = await response.json();
      data.status = 200;
      return data;
    } catch (err) {
      console.log("catch err",err)
      return err;
    }
  } catch (error) {
    console.error("API call failed:", error.message);
    throw error;
  }
};
```

### Current vs. Expected behavior

**Current**

Api is failing in server component. But working in client components. And this is random, sometimes API is working in server component as well.
<img width="683" alt="image" src="https://github.com/user-attachments/assets/284738ea-7cd2-4d89-9bf8-15be74678722">

### Provide environment information

```bash
next: "14.2.9"
node: v18.20.3
npm: v10.7.0
macOS: 14.5
```

### Which area(s) are affected? (Select all that apply)

Developer Experience, Runtime

### Which stage(s) are affected? (Select all that apply)

next dev (local), Vercel (Deployed)

### Additional context

API failing randomly in sever component, means it works sometimes and fails sometimes but when i checked same api in client component, it is working every time

This is the api : https://pepsindia.in/api/get-home-content

**Comment by samcx** [comment]
@vinitgundeti I am not able to replicate this issue (it's unlikely that it's related to Next.js).

How often are you experiencing this issue?

**Reply by vinitgundeti** [reply]
@samcx Experiencing this issue randomly. It works fine and sometimes suddenly it gives error. and it again works fine after restarting the system.

**Reply by ethicalhub** [reply]
> @samcx Experiencing this issue randomly. It works fine and sometimes suddenly it gives error. and it again works fine after restarting the system.

https://nextjs-faq.com/fetch-api-in-rsc

**Reply by vinitgundeti** [reply]
Hi @samcx this is not related to next js. There was some issue in domain pointing in Api itself.
